obj/objects.cpp

```diff
@@ -271,7 +281,10 @@ namespace objects
         if (point1_ == point2_ || point1_ == point3_ || point2_ == point3_)
```

Comment:
Isn't it better to check for the area of the triangle instead? It will be much more robust to floating point issues. You can have a triangle that passes this check, but its area will be so tiny that you may screw up entire computation.

Reply:
true, I will fix that in a minute

Reply:
i cant replace this with area check for now because it requires a lot of work. I will add another else if and add TODO, to change this in the future ;)

Reply:
Why does it requires a lot of work?

Next time if you say some statement, please try to elaborate. Things that are obvious to you may not be obvious to everyone.

Also if "a lot of work" means "I changed it and tests failed", then this is **not** a good reason to skip it. Remember you mentioned you changed double -> float and got 1032131231 errors, but it was not that hard to fix in practice, yet you added a TODO.

Reply:
it works now ;)

Reply:
You did not answer the question. What was this "a lot of work" you did? I can't see it.

Reply:
it turned out that it wasnt a big deal.... I just needed to add this->recalulateArea at the top of the method arePointsValid() and it works ;) the issue was that some things that I was calulateing later requeires to have points already valid, and when arepointsValid() method was called, area was still uninitialized ;)

Reply:
Thanks!

So then why did you say it was a lot of work? ;-)

Regardless, issue you are describing seems like a sign of bad class design.

Reply:
yep, I need to fix the class design, because i'm calling recalulateArea() twice, which is not necessary